pynvjitlink/patch.py

```diff
@@ -44,6 +48,24 @@
     Linker = object
 
 
+spec = importlib.util.find_spec("numba_cuda")
+if spec is not None:
+    _numba_cuda_in_use = True
+    _numba_cuda_error = "`numba_cuda` includes patches from pynvjitlink, so no further patches are needed. "
+
+    import numba_cuda
+
+    numba_cuda_ver = (int(x) for x in numba_cuda.__version__.split("."))
```

Comment:
On Python 3.11 at least, this results in comparing a generator with a tuple, which isn't supported:

```
In [1]: from pynvjitlink import patch
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 from pynvjitlink import patch

File ~/numbadev/pynvjitlink/pynvjitlink/patch.py:59
     56 import numba_cuda
     58 numba_cuda_ver = (int(x) for x in numba_cuda.__version__.split("."))
---> 59 if numba_cuda_ver < (0, 2, 0):
     60     suggestion = "Instead, use NUMBA_CUDA_ENABLE_PYNVJITLINK environment variable to enable pynvjitlink features."
     61 else:

TypeError: '<' not supported between instances of 'generator' and 'tuple'
```

One way around this seems to be:

```suggestion
    numba_cuda_ver = tuple(int(x) for x in numba_cuda.__version__.split("."))
```

Reply:
I've had success with `looseversion` for this type of situation, might be worth depending on for ease of use and futureproofing.